Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/random_perturbation.h

```diff
@@ -0,0 +1,172 @@
+        const Point_3 closest = tree.closest_point(np); //project on input surface
+
+        put(vpmap, v, closest);
+      }
+    }
+  }
+
+} //end namespace internal
+
+/*!
+* \ingroup PMP_meshing_grp
+* @brief randomly perturbs the locations of vertices of a triangulated surface mesh.
+* Note that depending on the chosen parameters, inversions and self-intersections may happen.
+*
+* @tparam TriangleMesh model of `MutableFaceGraph`.
+* @tparam NamedParameters a sequence of \ref namedparameters
+*
+* @param tmesh the triangulated surface mesh to be perturbed
+* @param perturbation_max_size the maximal length of moves that can be applied to
+*        vertices of `tmesh`.
+* @param np optional sequence of \ref namedparameters among the ones listed below
+*
+* \cgalNamedParamsBegin
+*  \cgalParamBegin{geom_traits} a geometric traits class instance, model of `Kernel`.
+*    Exact constructions kernels are not supported by this function.
+*  \cgalParamEnd
+*  \cgalParamBegin{vertex_point_map} the property map with the points associated
+*    to the vertices of `tmesh`. Instance of a class model of `ReadWritePropertyMap`.
+*  \cgalParamEnd
+*  \cgalParamBegin{vertex_is_constrained_map} a property map containing the
+*    constrained-or-not status of each vertex of `tmesh`. A constrained vertex
+*    cannot be modified at all during remeshing
+*  \cgalParamEnd
+* \cgalNamedParamsEnd
+*
+*/
+template<typename TriangleMesh, typename NamedParameters>
+void random_perturbation(TriangleMesh& tmesh
+                       , const double& perturbation_max_size
+                       , const NamedParameters& np)
+{
+  typedef TriangleMesh PM;
+  using boost::get_param;
+  using boost::choose_param;
+
+  typedef typename boost::graph_traits<PM>::vertex_descriptor vertex_descriptor;
+
+#ifdef CGAL_PMP_RANDOM_PERTURBATION_VERBOSE
+  std::cout << std::endl;
+  CGAL::Timer t;
+  std::cout << "Random perturbation (max size = "<< perturbation_max_size<<")...";
+  std::cout.flush();
+  t.start();
+#endif
+
+  typedef typename GetGeomTraits<PM, NamedParameters>::type GT;
+
+  typedef typename GetVertexPointMap<PM, NamedParameters>::type VPMap;
+  VPMap vpmap = choose_param(get_param(np, internal_np::vertex_point),
+                             get_property_map(vertex_point, tmesh));
+
+  typedef typename boost::lookup_named_param_def <
+      internal_np::vertex_is_constrained_t,
+      NamedParameters,
+      internal::No_constraint_pmap<vertex_descriptor>//default
+    > ::type VCMap;
+  VCMap vcmap = choose_param(get_param(np, internal_np::vertex_is_constrained),
+                             internal::No_constraint_pmap<vertex_descriptor>());
+
+  internal::random_perturbation_impl<GT>(tmesh,
+                                         perturbation_max_size,
+                                         vcmap,
+                                         vpmap,
+                                         0/*seed*/);
```

Comment:
with commit a874473106dba839e603a85cad6dfd8b9c9161e4, the seed becomes a named parameter